grant-project/zcash-grant-system
# Zcash e2e basics

Closes #201.

I'm hoping the backend e2e additions are not too gross. LMK if there are any concerns.

### What this does
- clears out remaining eth tests & code
- adds BE .env `E2E_TESTING` and `E2E_DATABASE_URL`
- adds `/api/v1/e2e` when `E2E_TESTING` is true, this has end-points for db setup, getting emails, and confirming contributions
- when in e2e mode, intercepts email sends and makes the last one available on the e2e api
- when in e2e mode, intercepts some of the blockchain api calls and returns mocks
- tests for: browsing, authentication, account recovery, and proposal funding

### What this doesn't do
- does not set up CI, but sets decent groundwork for it
- many more tests should be written (nothing for admin yet)


### Testing
- should be able to run without blockchain-watcher running
- make sure frontend is running
- create a postgres database for e2e, such as `zcash-e2e`
- set `/backend/.env`:
    ```bash
    E2E_TESTING=true
    E2E_DATABASE_URL="postgres://postgres:<PASSWORD>@localhost:5432/<E2E_DB_NAME>"
    ```
- start the backend
- `cd e2e`
- `rm -rf node_modules` (cypress has been upgraded, and web3 deps removed)
- `yarn`
-  `yarn cypress:open`
- wait until the cypress gui opens, then click "Run all specs"
    <img width="348" alt="Screen Shot 2019-03-12 at 12 31 03 AM" src="https://user-images.githubusercontent.com/11430954/54176777-23d09e80-445e-11e9-9ab9-9ff56e122bdb.png">
- all tests should pass
